### PR TITLE
test providing fallback in case of kernel fusion failure in MOC computation

### PR DESCRIPTION
**Pull request by grlee77**
testing potential fix for the `test_moc` [failure](https://github.com/rapidsai/cucim/actions/runs/5539741155/jobs/10144657106) seen in #572

### CODE REVIEW

**Comment by jakirkham** [comment]
Also more details in this comment: https://github.com/rapidsai/cucim/pull/572#issuecomment-1631610100

File: python/cucim/src/cucim/skimage/measure/_colocalization.py [hunk 1]
```diff
@@ -261,7 +269,15 @@ def manders_overlap_coeff(image0, image1, mask=None):
...
-    return _get_manders_overlap_coeff(image0, image1)
+    try:
+        return _get_manders_overlap_coeff(image0, image1)
+    except driver.CUDADriverError:
+        # One case on CI recently failed to fuse kernels. I was unable to
+        # reproduce this locally with the same NumPy and CuPy versions.
+        warnings.warn(
+            "kernel fusion failed, falling back to non-fused implementation"
+        )
+        return _get_manders_overlap_coeff_no_fusion(image0, image1)
```
Comment (jakirkham) [inline python/cucim/src/cucim/skimage/measure/_colocalization.py:280]: cc @leofang (in case you have insight into what causes this to fail on CUDA 11.2)

Comment (leofang) [reply python/cucim/src/cucim/skimage/measure/_colocalization.py:280]: What's the GPU & driver version in use, do you know @jakirkham ?

Comment (jakirkham) [reply python/cucim/src/cucim/skimage/measure/_colocalization.py:280]: From [one CI log]( https://github.com/rapidsai/cucim/actions/runs/5524163077/jobs/10078441622?pr=572#step:7:710 ) showing this issue, am seeing the following:

<details>

```
Tue Jul 11 21:46:39 2023       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 470.182.03   Driver Version: 470.182.03   CUDA Version: 11.4     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla V100-SXM2...  On   | 00000000:0A:00.0 Off |                    0 |
| N/A   34C    P0    42W / 300W |      0MiB / 32510MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
```

</details>

File: python/cucim/src/cucim/skimage/measure/_colocalization.py [hunk 2]
```diff
@@ -182,6 +185,11 @@ def _get_manders_overlap_coeff(image0, image1):
     return cp.sum(cp.multiply(image0, image1)) / denom
 
 
+def _get_manders_overlap_coeff_no_fusion(image0, image1):
+    denom = (cp.sum(cp.square(image0)) * (cp.sum(cp.square(image1)))) ** 0.5
+    return cp.sum(cp.multiply(image0, image1)) / denom
```
Comment (jakirkham) [inline python/cucim/src/cucim/skimage/measure/_colocalization.py:190]: Wonder if we can just write this in terms of linalg operations and benefit from BLAS optimizations under-the-hood. Tried this in PR ( https://github.com/rapidsai/cucim/pull/578 ). If we do that, do we still need `fuse`?

Comment (grlee77) [reply python/cucim/src/cucim/skimage/measure/_colocalization.py:190]: Yes, that sounds great!

**Comment by jakirkham** [comment]
Thanks Greg! 🙏

**Pull request closed without merging**